src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -472,5 +1117,158 @@ internal static unsafe float Dist2(Span<float> src, Span<float> dst)
             }
         }
 
+        internal static unsafe void ZeroItemsU(AlignedArray dst, int c, int[] indices, int cindices)
+        {
+            fixed (float* pDstStart = &dst.Items[0])
+            fixed (int* pidx = &indices[0])
+            {
+                float* pdst = Ptr(dst, pDstStart);
+
+                // REVIEW NEEDED: This line expands to (void)(c); but is it necessary?
+                // DEBUG_ONLY(c);
+
+                for (int i = 0; i < cindices; ++i)
+                {
+                    int index = pidx[i];
+                    Contracts.Assert(0 <= index && index < c);
+                    pdst[index] = 0;
+                }
+            }
+        }
+
+        internal static unsafe void ZeroMatrixItemsCore(AlignedArray dst, int c, int ccol, int cfltRow, int[] indices, int cindices)
+        {
+            fixed (float* pDstStart = &dst.Items[0])
+            fixed (int* pidx = &indices[0])
+            {
+                float* pdst = Ptr(dst, pDstStart);
+
+                // REVIEW NEEDED: This line expands to (void)(c); but is it necessary?
+                // DEBUG_ONLY(c);
+
+                int ivLogMin = 0;
+                int ivLogLim = ccol;
+                int ivPhyMin = 0;
+
+                for (int i = 0; i < cindices; ++i)
+                {
+                    int index = pidx[i];
+                    Contracts.Assert(0 <= index && index < c);
+
+                    int col = index - ivLogMin;
+                    if ((uint)col >= (uint)ccol)
+                    {
+                        Contracts.Assert(ivLogMin > index || index >= ivLogLim);
+
+                        int row = index / ccol;
+                        ivLogMin = row * ccol;
+                        ivLogLim = ivLogMin + ccol;
+                        ivPhyMin = row * cfltRow;
+
+                        Contracts.Assert(ivLogMin <= index && index < ivLogLim);
+                        col = index - ivLogMin;
+                    }
+
+                    pdst[ivPhyMin + col] = 0;
+                }
+            }
+        }
+
+        internal static unsafe void SdcaL1UpdateU(float primalUpdate, Span<float> src, float threshold, Span<float> v, Span<float> w)
+        {
+            fixed (float* psrc = src)
+            fixed (float* pdst1 = v)
+            fixed (float* pdst2 = w)
+            {
+                float* pSrcEnd = psrc + src.Length;
+                float* pSrcCurrent = psrc;
+                float* pDst1Current = pdst1;
+                float* pDst2Current = pdst2;
+
+                Vector128<float> xPrimal = Sse.SetAllVector128(primalUpdate);
+
+                Vector128<float> signMask = Sse.SetAllVector128(-0.0f); // 1000 0000 ...
```

Comment:
(nit) I know this is ported from the C++ code, but I think the comment is slightly confusing/misleading.  typically I would expect hexadecimal representation here `0x8000 0000`.  I guess this is binary representation, which is why it is using the `...`.

it may make sense to call this out as binary using the `0b` prefix.  Or my preference would be just to use hex - `0x8000 0000`.